cpp/src/arrow/compute/kernels/vector_selection_take_internal.cc

```diff
@@ -395,10 +526,45 @@ struct FixedWidthTakeImpl {
     out_arr->null_count = out_arr->length - valid_count;
     return Status::OK();
   }
+
+  static Status ChunkedExec(KernelContext* ctx, const ChunkedArray& values,
+                            const ArraySpan& indices, ArrayData* out_arr,
+                            int64_t factor) {
+    const bool out_has_validity = values.null_count() > 0 || indices.MayHaveNulls();
```

Comment:
As an optimization, do we want to fallback on ArraySpan-take if `values.num_chunks() == 1`?

Reply:
Or it seems this is actually handled at a higher level, in the kernel exec scaffolding?

Reply:
`SpecialTakeChunkedExec` takes care of this. Look for the `single_chunk` variable. It will call `take_aaa_exec` instead of `take_caa_exec` and then produce a chunked array output with a single chunk.